### PR TITLE
Light update to Flax NNX Model Surgery

### PR DESCRIPTION
**Pull request by 8bitmp3**
@cgarciae @IvyZX 

preview: https://flax--4276.org.readthedocs.build/en/4276/guides/surgery.html

### CODE REVIEW

File: docs_nnx/guides/surgery.ipynb [hunk 1]
```diff
@@ -6,15 +6,13 @@
...
     "# Model surgery\n",
     "\n",
-    "> **Attention**: This page relates to the new Flax NNX API.\n",
+    "In this guide, you will learn how to perform model surgery in Flax NNX using several real-world scenarios:\n",
     "\n",
-    "In this guide you will learn how to do model surgery with Flax NNX with several real-scenario use cases:\n",
+    "* __Pythonic `nnx.Module` manipulation__: Using Pythonic ways to manipulate sub-`Module`s given a model.\n",
     "\n",
-    "* __Pythonic module manipulation__: Pythonic ways to manipulate sub-modules given a model.\n",
+    "* __Manipulation of an abstract model or state__: A key trick for playing with `flax.nnxModule`s and states without memory allocation.\n",
```
Comment (IvyZX) [inline docs_nnx/guides/surgery.ipynb:13]: nit: `flax.nnxModule` -> `flax.nnx.Module`

File: docs_nnx/guides/surgery.ipynb [hunk 2]
```diff
@@ -164,7 +166,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "When you fill every `VariableState` leaf's `value`s with real jax arrays, the abstract model becomes equivalent to a real model."
+    "When you fill every `nnx.VariableState` pytree leaf's `value`s with real `jax.Array`s, the abstract model becomes equivalent to a real model."
```
Comment (IvyZX) [inline docs_nnx/guides/surgery.ipynb:169]: `value`s  ->  `value` attributes

File: docs_nnx/guides/surgery.ipynb [hunk 3]
```diff
@@ -209,7 +213,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "In this new model, the sub-modules are renamed from `linear(1|2)` to `layer(1|2)`. Since the pytree structure changed, it's impossible to load the old checkpoint with the new model state structure:"
+    "In this new model, the sub-`Module`s are renamed from `linear(1|2)` to `layer(1|2)`. Since the pytree structure has changed, it is impossible to load the old checkpoint with the new model state structure:"
```
Comment (IvyZX) [inline docs_nnx/guides/surgery.ipynb:216]: "it is impossible to load" -> "it is impossible to directly load"

File: docs_nnx/guides/surgery.ipynb [hunk 4]
```diff
@@ -318,9 +325,9 @@
...
+    "To do naive partial initialization, you can just initialize the whole model, then swap the pre-trained parameters in. However, this approach may allocate additional memory midway if your modification requires re-creating module parameters that you will later discard. Below is an example of this.\n",
     "\n",
-    "> Note: You can use `jax.live_arrays()` to check all the arrays live in memory at any given time. This call can be messed up when you run a single notebook cell multiple times (due to garbage-collecting old python variables), but restarting the kernel and running from scratch will always yield same output."
+    "> **Note:** You can use `jax.live_arrays()` to check all the arrays live in memory at any given time. This call can be “messed up” when you run a single Jupyter notebook cell multiple times (because of old garbage-collecting Python variables). However, restarting the Python kernel in the notebook and running the code from scratch will always yield the same output."
```
Comment (IvyZX) [inline docs_nnx/guides/surgery.ipynb:330]: "because of old garbage-collecting Python variables" -> "due to garbage-collection of old Python variables"

**Review by IvyZX — APPROVED** [review]
Thank you! Just some nits.

File: docs_nnx/guides/surgery.ipynb (hunk 4 above)
Comment (8bitmp3) [reply docs_nnx/guides/surgery.ipynb:330]: good catch, thanks! done

File: docs_nnx/guides/surgery.ipynb (hunk 3 above)
Comment (8bitmp3) [reply docs_nnx/guides/surgery.ipynb:216]: thanks, done!

File: docs_nnx/guides/surgery.ipynb (hunk 2 above)
Comment (8bitmp3) [reply docs_nnx/guides/surgery.ipynb:169]: Nice, done!

File: docs_nnx/guides/surgery.ipynb (hunk 1 above)
Comment (8bitmp3) [reply docs_nnx/guides/surgery.ipynb:13]: Good catch, done! Thanks!

**Pull request merged**